### DISCUSSION TITLE
Left drawer with nicegui

### DISCUSSION THREAD

**Discussion by dega5** [zauberzeug/nicegui#1883] [Q&A] [answered]
### Question

I am trying to reduce the width of my left drawer to 225px but can't with nicegui

```py
with ui.label().classes(replace=' width: 225px !important;'):
         with ui.left_drawer(top_corner=False, bottom_corner=False).style(replace='height: calc(100% + 20px) !important;  background-color: #f3f3f3; border-right: 1px solid #ccc; width: 225px !important;').props('width:225'):
            with ui.column().style('padding: 10px;'):
                with ui.column():
                    ui.link('Link 1 ', '/').classes(replace='text-lg')
                with ui.column():
                    ui.link('Link 2', '/').classes(replace='text-lg') 
```

This is what it looks like right now (default 300px)
![08_01_31](https://github.com/zauberzeug/nicegui/assets/96119834/362bd3e9-8ad7-4578-af7c-96f45bca4106)

I saw this line of code while inspecting element, but cannot see it anywhere in my nicegui code and I am not sure how to change it.
![08_01_46](https://github.com/zauberzeug/nicegui/assets/96119834/7c6a1116-d1e2-48e6-ab72-7bbeacac0f56)

I came across this article where the default size of drawer is 300 but I am not sure how to use it for my code.

https://quasar.dev/layout/drawer#layout-builder

Would appreciate any help!

**Comment by falkoschindler** [comment, marked as answer]
Using QDrawer's props is the right approach. But prop values need to be set with an equal sign:
```py
with ui.left_drawer(bordered=True).props('width=225'):
    ui.label('Label 1')
    ui.label('Label 2')
```

**Reply by v479038280** [reply]
Mark as a favorite !

**Reply by dega5** [reply]
Thank you so much!! Such a simple solution but spent so much time on it :(

**Reply by FLandolt** [reply]
HI sorry maybe an additional question to this and not to hard can i somehow set the drawer width to half the page width ?

**Reply by falkoschindler** [reply]
@FLandolt I guess you need some custom CSS:
```py
ui.add_head_html('<style>.q-drawer--left { width: 50% !important }</style>')
```

**Reply by FLandolt** [reply]
Thanks a lot. It worked like a charm!

**Reply by BejanSadeghian** [reply]
> @FLandolt I guess you need some custom CSS:
> 
> ```python
> ui.add_head_html('<style>.q-drawer--left { width: 50% !important }</style>')
> ```

This had the body elements respect when the drawer is not collapsed and resize when toggled. Thank you. 

The top answer ended up having elements show up behind the drawer for me. Not sure why but that wasnt a desired outcome for me.
